Source/EasyNetQ/Topology/ExchangeToExchangeBinding.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace EasyNetQ.Topology
+{
+    /// <summary>
+    ///     Binding between exchange and exchange
+    /// </summary>
+    public readonly struct ExchangeToExchangeBinding
+    {
+        /// <summary>
+        ///     Creates Binding
+        /// </summary>
+        public ExchangeToExchangeBinding(Exchange source, Exchange destination, string routingKey, IDictionary<string, object> arguments = null)
+        {
+            Preconditions.CheckNotNull(routingKey, "routingKey");
+
+            Destination = destination;
+            Source = source;
+            RoutingKey = routingKey;
+            Arguments = arguments == null ? null : new ReadOnlyDictionary<string, object>(arguments);
```

Comment:
Static ReadOnlyDictionary from field.

Reply:
@sungam3r sorry I didn't get it, could you explain?

Reply:
@micdenny You are not alone 🤣

Reply:
I've re-read and it looks like the suggestion is to return a statically cached empty dictionary instead of null.

Reply:
ok i've read above

Reply:
I have doubts about `returning a statically cached empty dictionary instead of null`. 

Current `RabbitMQ.Client` forces to use null instead of empty dictionaries to reduce allocations(also we do the same in *Configuration classes). Despite the fact that it's a breaking change I suppose that it unifies the way how dictionaries used in our codebase.

Reply:
but a static dictionary should not consume too much memory, before was a new instance every time, and if we use a static dictionary we can improve the memory and avoid a possible breaking change

Reply:
Sure thing, but I'm mostly about a confusion about to do null-check or not

Reply:
> Current RabbitMQ.Client forces to use null instead of empty dictionaries 

Well, 50/50, so do as you see fit.